source/common/upstream/outlier_detection_impl.cc

```diff
@@ -471,8 +478,24 @@ void DetectorImpl::ejectHost(HostSharedPtr host,
           runtime_.snapshot().getInteger(BaseEjectionTimeMsRuntime, config_.baseEjectionTimeMs()));
       const std::chrono::milliseconds max_eject_time = std::chrono::milliseconds(
           runtime_.snapshot().getInteger(MaxEjectionTimeMsRuntime, config_.maxEjectionTimeMs()));
-      if ((host_monitors_[host]->ejectTimeBackoff() * base_eject_time) <
-          (max_eject_time + base_eject_time)) {
+      const std::chrono::milliseconds max_eject_time_jitter = std::chrono::milliseconds(
+          runtime_.snapshot().getInteger(MaxEjectionTimeJitterMsRuntime, config_.maxEjectionTimeJitterMs()));
+
+      // Generate random jitter so that not all hosts uneject at the same time, 
+      // which could possibly generate a connection storm
+
+      // std::chrono::milliseconds receives a signed integer as input, so we ensure that our random
+      // random value is in the range [0, std::numeric_limits<long>::max())
+      long random_long = static_cast<long>(random_generator_() % (std::numeric_limits<long>::max()));
+      // cap the jitter at max_eject_time_jitter
+      const std::chrono::milliseconds jitter =
+          std::chrono::milliseconds(random_long) % (max_eject_time_jitter + std::chrono::milliseconds(1));
```

Comment:
Not sure if it's intended or not, but this contradicts the API comment that the default jitter is 0s.
To avoid breaking compatibility, the value of `jitter` should be 0, and only if `max_ejection_time_jitter_ms_` is non-zero, this code should set the random value of `jitter`.

Reply:
When max_ejection_time_jitter is not set, it's value is 0. Then the jitter will be: random_long % 1 and will always equal 0.